### PR TITLE
En 4717 termui cpu load

### PR DESCRIPTION
**Pull request by miiu96**
- Fix in termui problem with CPU load
-Update gopsutil library

### CODE REVIEW

File: core/statistics/machine/cpuStatistics.go [hunk 1]
```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
...
-		cs.setZeroStatsAndWait()
 		return
 	}
 
-	numPhysicalCores, err := cpu.Counts(false)
-	if err != nil {
-		cs.setZeroStatsAndWait()
-		return
-	}
+	cpuUsagePercent := percent / float64(cs.numCpu)
```
Comment (SebastianMarian) [inline core/statistics/machine/cpuStatistics.go]: Is there a chance that cs.numCpu to be 0. Maybe you can check this and if it is 0 you can set cpuUsagePercent to 100

File: core/statistics/machine/cpuStatistics.go [hunk 2]
```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
...
 
-	cpuUsagePercent = cpuUsagePercent / float64(numPhysicalCores)
 	atomic.StoreUint64(&cs.cpuPercentUsage, uint64(cpuUsagePercent))
```
Comment (SebastianMarian) [inline core/statistics/machine/cpuStatistics.go]: I would rename cs.cpuPercentUsage in cs.cpuUsagePercent

File: core/statistics/machine/cpuStatistics.go [hunk 3]
```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
...
-	time.Sleep(durationSecond)
 }
 
 func (cs *CpuStatistics) setZeroStatsAndWait() {
```
Comment (SebastianMarian) [inline core/statistics/machine/cpuStatistics.go]: Is there this method used anymore? If not, you can remove it

File: core/statistics/machine/cpuStatistics.go [hunk 4]
```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
 
 // CpuStatistics can compute the cpu usage percent
 type CpuStatistics struct {
+	numCpu          int
 	cpuPercentUsage uint64
 }
 
+// NewCpuStatistics will create a CpuStatistics object
+func NewCpuStatistics() *CpuStatistics {
+	numCpu, _ := cpu.Counts(true)
```
Comment (iulianpascalau) [inline core/statistics/machine/cpuStatistics.go]: You are hiding error here. Must be careful with these things, as mentioned by Sebastian that this function might return in some situations numCpu = 0 and error. Your code will break in panic on line 41.
The constructor might return (*CpuStatistics, error) tuple.

File: core/statistics/machine/cpuStatistics.go (hunk 1 above)
Comment (iulianpascalau) [reply core/statistics/machine/cpuStatistics.go]: We need to test this on all archs/oses (linux, mac, windows, raspPi)

File: core/statistics/machine/cpuStatistics.go (hunk 2 above)
Comment (miiu96) [reply core/statistics/machine/cpuStatistics.go]: Done.

File: core/statistics/machine/cpuStatistics.go (hunk 3 above)
Comment (miiu96) [reply core/statistics/machine/cpuStatistics.go]: Done. Method removed.

File: core/statistics/machine/cpuStatistics.go (hunk 4 above)
Comment (miiu96) [reply core/statistics/machine/cpuStatistics.go]: Done.

File: core/statistics/machine/cpuStatistics.go (hunk 1 above)
Comment (miiu96) [reply core/statistics/machine/cpuStatistics.go]: Done. Works on all that 4 platforms.

Comment (iulianpascalau) [reply core/statistics/machine/cpuStatistics.go]: ok

**Pull request merged**